extensions/broker-server-api/client/build.gradle.kts

```diff
@@ -95,14 +98,31 @@ checkstyle {
 
 
 tasks.getByName<JavaCompile>("compileJava") {
-    dependsOn("postprocessGeneratedClient")
+    dependsOn(postprocessGeneratedClient)
+}
+
+val sourcesJar by tasks.creating(Jar::class) {
+    dependsOn(postprocessGeneratedClient)
+    dependsOn(JavaPlugin.CLASSES_TASK_NAME)
+    archiveClassifier.set("sources")
+    from(sourceSets["main"].allSource)
+}
+
+val javadocJar by tasks.creating(Jar::class) {
+    dependsOn(postprocessGeneratedClient)
+    dependsOn(JavaPlugin.JAVADOC_TASK_NAME)
+    archiveClassifier.set("javadoc")
+    from(tasks["javadoc"])
+}
+
+artifacts {
+    add("archives", sourcesJar)
+    add("archives", javadocJar)
 }
```

Comment:
can't this be handled in the submodules clause of the root gradle file?

Reply:
It could, but if I put it in the root gradle, all the projects will depend on `postprocessGeneratedClient`, which is API-wrapper specific.

Reply:
These code blocks are task adjustments, so they are additive, right?

What I meant, is, that we could enable source jars for all jars by moving this to the submodules directive:


```kt
// root build.gradle.kts submodules section
val sourcesJar by tasks.creating(Jar::class) {
    dependsOn(JavaPlugin.CLASSES_TASK_NAME)
    archiveClassifier.set("sources")
    from(sourceSets["main"].allSource)
}

val javadocJar by tasks.creating(Jar::class) {
    dependsOn(JavaPlugin.JAVADOC_TASK_NAME)
    archiveClassifier.set("javadoc")
    from(tasks["javadoc"])
}
```

maybe in the root just the default withSourcesJar() would be sufficient

```kt
// api wrapper build.gradle.kts
val sourcesJar by tasks.creating(Jar::class) {
    dependsOn(postprocessGeneratedClient)
}

val javadocJar by tasks.creating(Jar::class) {
    dependsOn(postprocessGeneratedClient)
}
```

Reply:
i meant in the bottom block where you "get" tasks of the given class / type / name to adjust them once more

Reply:
ok, got it. But there's an shorter way, see
https://github.com/sovity/edc-extensions/pull/921/files#diff-c0dfa6bc7a8685217f70a860145fbdf416d449eaff052fa28352c5cec1a98c06R125

Reply:
And now we have sources and javadocs for all the deliveries. ^_^